tests/test_button.py

```diff
@@ -0,0 +1,104 @@
+
+    entry = entity_registry.async_get("button.my_model_s_horn")
+    assert entry.unique_id == "tesla_model_s_111111_horn"
+
+    entry = entity_registry.async_get("button.my_model_s_flash_lights")
+    assert entry.unique_id == "tesla_model_s_111111_flash_lights"
+
+    entry = entity_registry.async_get("button.my_model_s_wake_up")
+    assert entry.unique_id == "tesla_model_s_111111_wake_up"
+
+    entry = entity_registry.async_get("button.my_model_s_force_data_update")
+    assert entry.unique_id == "tesla_model_s_111111_force_data_update"
+
+    entry = entity_registry.async_get("button.my_model_s_trigger_homelink")
+    assert entry.unique_id == "tesla_model_s_111111_trigger_homelink"
+
+
+async def test_horn_press(hass: HomeAssistant) -> None:
+    """Tests car horn button press."""
+    await setup_platform(hass, BUTTON_DOMAIN)
+
+    with patch("teslajsonpy.car.TeslaCar.honk_horn") as mock_honk_horn:
+        assert await hass.services.async_call(
+            BUTTON_DOMAIN,
+            "press",
+            {ATTR_ENTITY_ID: "button.my_model_s_horn"},
+            blocking=True,
+        )
+        mock_honk_horn.assert_awaited_once()
+
+
+async def test_flash_lights_press(hass: HomeAssistant) -> None:
+    """Tests car flash lights button press."""
+    await setup_platform(hass, BUTTON_DOMAIN)
+
+    with patch("teslajsonpy.car.TeslaCar.flash_lights") as mock_flash_lights:
+        assert await hass.services.async_call(
+            BUTTON_DOMAIN,
+            "press",
+            {ATTR_ENTITY_ID: "button.my_model_s_flash_lights"},
+            blocking=True,
+        )
+        mock_flash_lights.assert_awaited_once()
+
+
+async def test_wake_up_press(hass: HomeAssistant) -> None:
+    """Tests car wake up button press."""
+    await setup_platform(hass, BUTTON_DOMAIN)
+
+    with patch("teslajsonpy.car.TeslaCar.wake_up") as mock_wake_up:
+        assert await hass.services.async_call(
+            BUTTON_DOMAIN,
+            "press",
+            {ATTR_ENTITY_ID: "button.my_model_s_wake_up"},
+            blocking=True,
+        )
+        mock_wake_up.assert_awaited_once()
+
+
+async def test_force_data_update_press(hass: HomeAssistant) -> None:
+    """Tests car force data button press."""
+    await setup_platform(hass, BUTTON_DOMAIN)
+
+    with patch(
+        "custom_components.tesla_custom.base.TeslaCarEntity.update_controller"
+    ) as mock_force_data_update:
+        assert await hass.services.async_call(
+            BUTTON_DOMAIN,
+            "press",
+            {ATTR_ENTITY_ID: "button.my_model_s_force_data_update"},
+            blocking=True,
+        )
+        mock_force_data_update.assert_awaited_once_with(wake_if_asleep=True, force=True)
+
+
+# async def test_trigger_homelink_press(hass: HomeAssistant) -> None:
+#     """Tests car trigger homelink button press."""
+#     await setup_platform(hass, BUTTON_DOMAIN)
+#     # Need a way to enable this device before running tests (disabled by default)
```

Comment:
There's a [pytest fixture in HA](https://github.com/home-assistant/core/blob/dev/tests/components/conftest.py#L28) that can be used. I tried copying it over to conftest.py in our test folder and added the `entity_registry_enabled_by_default` parameter to this test but it still doesn't seem to be enabling the device. Any ideas?

Reply:
I ended up just adding a check for the HomeLink button and only adding it if a car has the capability.